avocado/utils/kernel.py

```diff
@@ -95,11 +99,26 @@ def build(self):
         Build kernel from source.
         """
         log.info("Starting build the kernel")
+        build_output_format = ""
+        if self.do_install is not None:
+            if self.distro.name == "Ubuntu":
+                build_output_format = "deb-pkg"
         if self.config_path is None:
-            build.make(self.linux_dir, extra_args='O=%s defconfig' % self.build_dir)
+            build.make(self.linux_dir, extra_args='-C %s defconfig' % self.linux_dir)
         else:
-            build.make(self.linux_dir, extra_args='O=%s olddefconfig' % self.build_dir)
-        build.make(self.linux_dir, extra_args='O=%s' % self.build_dir)
+            build.make(self.linux_dir, extra_args='-C %s olddefconfig' % self.linux_dir)
+        build.make(self.linux_dir, extra_args='-C %s %s' % (self.linux_dir, build_output_format))
```

Comment:
IIUC it correctly, you can just build the usual kernel (the default target) on `build()`, and then, on install, you can build the distro specific target (`build_output_format`), which will reuse the previous work done on `build()`.

Reply:
Yes; that is correct.